fhirpathpy/engine/invocations/navigation.py

```diff
@@ -21,35 +21,52 @@ def func(acc, res):
                 value = data[prop]
                 childPath = ""
 
+                # extensions shouldn't filter through here, yet they should for descendants?
+                # unless this item is the node that is being processed (primitive extension)
+                # though if you filter it, descendants will not work too
+                if prop.startswith("_") and exclude_primitive_extensions:
+                    continue
+
                 if res.path is not None:
                     childPath = res.path + "." + prop
 
+                fullPath = f"{res.propName}.{prop}" if res.propName else childPath # The full path to the node (weill evenutally be) e.g. Patient.name[0].given
+                fullPath = fullPath.replace("_", "")
+
+                if prop == "extension":
+                    childPath = "Extension"
+
                 if (
                     isinstance(model, dict)
                     and "pathsDefinedElsewhere" in model
                     and childPath in model["pathsDefinedElsewhere"]
                 ):
                     childPath = model["pathsDefinedElsewhere"][childPath]
 
+                childPath = (
+                    model["path2Type"].get(childPath, childPath)
+                    if isinstance(model, dict) and "path2Type" in model
+                    else childPath
+                )
+                
                 if isinstance(value, list):
-                    mapped = [create_node(n, childPath) for n in value]
+                    mapped = [create_node(n, childPath, propName=f"{fullPath}[{i}]") for i, n in enumerate(value)]
```

Comment:
I saw that index was passed in the previous usages, but not there, was it done intentionally?